### PR TITLE
fix: self referencing relations

### PR DESCRIPTION
**Pull request by Marc-Roig**
### What does it do?

Fixes an issue only reproducible in Postgres and on new Strapi apps (not in the monorepo itself)

The database relation stealing was failing, and was due to the knex connection being "reused" to create subqueries.

```ts
// Before
connection
	.from(tableName)
	.where(
		'column', 
	     connection.select(...)  // Reused same connection
) 

// After
connection
	.from(tableName)
	.where(
		'column', 
	     (con) => con.select(...)  // New connection reference
) 

```

Aparently the former created a self reference inside the connection query

### CODE REVIEW

File: packages/core/database/src/entity-manager/regular-relations.ts [hunk 1]
```diff
@@ -83,7 +80,11 @@ const deletePreviousOneToAnyRelations = async ({
     .delete()
     .from(joinTable.name)
     // Exclude the ids of the current document
-    .whereNotIn(joinColumn.name, getDocumentSiblingIdsQuery(con, joinColumn.referencedTable!, id))
+    .whereNotIn(
+      // @ts-expect-error - knex incorrectly expects a string array
+      joinColumn.name,
```
Comment (Marc-Roig) [inline packages/core/database/src/entity-manager/regular-relations.ts:85]: @alexandrebodin  only when the second argument is a function, knex expects the first one to be an array of strings, do you know if that should be the case? This specific setup seems to work 🤷‍♂️  and the documentation doesn't mention anything about that either

**Comment by Marc-Roig** [comment]
I am going to create an experimental to validate the fix

**Review by Convly — COMMENTED** [review]
What a nice timing :smiley: 

https://github.com/strapi/strapi/pull/20543/files#r1645584094

Depending on whether you want to include the transactions support, we could merge our work here or in the other PR

File: packages/core/database/src/entity-manager/regular-relations.ts (hunk 1 above)
Comment (Convly) [reply packages/core/database/src/entity-manager/regular-relations.ts:85]: That's weird, I didn't have the issue in my PR :thinking:

**Comment by Marc-Roig** [comment]
@Convly oops, indeed. Lets see who can merge first, if I merge this first I can fix the conflicts on your branch 🚀

File: packages/core/database/src/entity-manager/regular-relations.ts (hunk 1 above)
Comment (Marc-Roig) [reply packages/core/database/src/entity-manager/regular-relations.ts:85]: this only happened to me if the second argument was a function (qb) => qb.subquery
When using a regular subquery , it worked fine 😞

Comment (alexandrebodin) [reply packages/core/database/src/entity-manager/regular-relations.ts:85]: super weird 🤔  

Another options is to build the query this where. not sure the typings would work better but who knows
```
 .where((builder) =>
    builder.whereIn('id', [1, 11, 15]).whereNotIn('id', [17, 19])
  )
```

**Comment by Marc-Roig** [comment]
@alexandrebodin I did require the func expressions, just that I made the `getDocumentSiblingIdsQuery` return a function instead, which makes it a bit cleaner

File: packages/core/database/src/entity-manager/regular-relations.ts (hunk 1 above)
Comment (Marc-Roig) [reply packages/core/database/src/entity-manager/regular-relations.ts:85]: fixes the type issue I do see the original max stack size error 🙃

**Comment by Marc-Roig** [comment]
ah!! missunderstood your comment :) yes! It didn't require using `this`

**Pull request merged**